packages/common/src/utils/functional.ts

```diff
@@ -24,8 +24,21 @@ export interface Compose {
 
 export const compose: Compose =
   (...transforms: Transform[]) =>
-  <T extends object>(value: T): T =>
-    [...transforms].reverse().reduce((acc, transform) => transform(acc), value);
+  <T extends object>(value: T): T => {
+    if (transforms.length === 0) {
+      return value;
+    }
+
+    if (transforms.length === 1) {
```

Comment:
Is there real cases where it makes sense to use just one (Maybe if we apply a transform conditionally)?
Wondering if we should possibly have a lint rule to avoid having just one.
I mean. `compose(double)(3)` should be  `double(3)` instead, right?

Reply:
yaeh, lint rule is a good, but useful during refactoring we forget  to remove compose if there's only one function, I guess we can remove the `=== 0` check, since we shouldn't have this case ever